File: app/src/test/java/net/tlalka/android/fiszki/models/entities/ClusterTest.java
package net.tlalka.android.fiszki.models.entities;

import net.tlalka.android.fiszki.models.types.OwnerType;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

public class ClusterTest {

    @Test
    public void testCheekDefaultConstructor() {

        // when
        Cluster cluster = new Cluster();

        // then
        assertNotNull(cluster);
        assertNull(cluster.getOwnerType());
        assertNull(cluster.getWords());
    }

    @Test
    public void testCheekCustomConstructor() {

        // given
        OwnerType user = OwnerType.USER;

        // when
        Cluster cluster = new Cluster(user);

        // then
        assertNotNull(cluster);
        assertNotNull(cluster.getOwnerType());
        assertEquals(user, cluster.getOwnerType());
    }
}

File: app/src/test/java/net/tlalka/android/fiszki/model/entities/ClusterTest.kt
package net.tlalka.android.fiszki.model.entities

import net.tlalka.android.fiszki.model.types.OwnerType
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Assert.assertNull
import org.junit.Test

class ClusterTest {

    @Test
    fun testCheekDefaultConstructor() {

        // when
        val cluster = Cluster()

        // then
        assertNotNull(cluster)
        assertEquals(OwnerType.SYSTEM, cluster.ownerType)
        assertNull(cluster.words)
    }

    @Test
    fun testCheekCustomConstructor() {

        // given
        val user = OwnerType.USER

        // when
        val cluster = Cluster(user)

        // then
        assertNotNull(cluster)
        assertNotNull(cluster.ownerType)
        assertEquals(user, cluster.ownerType)
    }
}
